openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/TypeGeneratorUtils.java

```diff
@@ -312,25 +313,31 @@ private static AnnotationNode generateArrayConstraint(ArraySchema arraySchema) {
 
     private static List<String> getNumberAnnotFields(Schema<?> numberSchema) {
         List<String> fields = new ArrayList<>();
-        if (numberSchema.getMinimum() != null && numberSchema.getExclusiveMinimum() == null) {
+
+        if ((numberSchema.getMinimum() != null) &&
+                (BigDecimal.ZERO.compareTo(numberSchema.getMinimum()) != 0)
+                && (numberSchema.getExclusiveMinimum() == null)) {
             String value = numberSchema.getMinimum().toString();
             String fieldRef = MINIMUM + COLON + value;
             fields.add(fieldRef);
         }
-        if (numberSchema.getMaximum() != null && numberSchema.getExclusiveMaximum() == null) {
+        if (numberSchema.getMaximum() != null &&
+                (BigDecimal.ZERO.compareTo(numberSchema.getMaximum()) != 0)
+                && numberSchema.getExclusiveMaximum() == null) {
             String value = numberSchema.getMaximum().toString();
             String fieldRef = MAXIMUM + COLON + value;
             fields.add(fieldRef);
         }
         if (numberSchema.getExclusiveMinimum() != null &&
-                numberSchema.getExclusiveMinimum() && numberSchema.getMaximum() != null) {
+                numberSchema.getExclusiveMinimum() && numberSchema.getMinimum() != null &&
+               (BigDecimal.ZERO.compareTo(numberSchema.getMinimum()) != 0)) {
             String value = numberSchema.getMinimum().toString();
             String fieldRef = EXCLUSIVE_MIN + COLON + value;
             fields.add(fieldRef);
         }
         if (numberSchema.getExclusiveMaximum() != null &&
                 numberSchema.getExclusiveMaximum() &&
-                numberSchema.getMaximum() != null) {
+                numberSchema.getMaximum() != null && (BigDecimal.ZERO.compareTo(numberSchema.getMaximum()) != 0)) {
```

Comment:
Can't we just use `numberSchema.getMaximum() != 0` in here?

Reply:
We cannot do it because `numberSchema.getMaximum() ` gives a big decimal value that does not support the `!=` operator.